File: src/main/java/com/dingdongdeng/coinautotrading/domain/entity/ExchangeKey.java
package com.dingdongdeng.coinautotrading.domain.entity;

import com.dingdongdeng.coinautotrading.common.type.CoinExchangeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@ToString
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "exchange_key")
public class ExchangeKey {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "exchange_key_id")
    private Long id;

    @Column(name = "pair_id")
    private String pairId;

    @Enumerated(EnumType.STRING)
    @Column(name = "coin_exchange_type")
    private CoinExchangeType coinExchangeType;

    @Column(name = "name")
    private String name;

    @Column(name = "value")
    private String value;

    @Column(name = "user_id")
    private String userId;

}

File: app-service/src/main/kotlin/com/dingdongdeng/autotrading/domain/exchange/entity/ExchangeKey.kt
package com.dingdongdeng.autotrading.domain.exchange.entity

import com.dingdongdeng.autotrading.infra.common.type.CoinExchangeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "exchange_key")
class ExchangeKey(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "exchange_key_id")
    val id: Long? = null,
    @Column(name = "pair_id")
    val pairId: String,
    @Enumerated(EnumType.STRING)
    @Column(name = "coin_exchange_type")
    val coinExchangeType: CoinExchangeType,
    @Column(name = "name")
    val name: String,
    @Column(name = "`value`")
    val value: String,
    @Column(name = "user_id")
    val userId: String,
)
